content/news/2021-04-04-bevy-0.5/index.md

```diff
@@ -0,0 +1,1254 @@
+ * Parallel systems:
+ -- "&app::increment_counter" and "&app::print_every_other_time"
+    conflicts: ["usize"]
+```
+
+The ambiguity detector found a conflict and mentions that adding an explicit dependency would resolve the conflict:
+
+```rust
+app
+    .add_system(increment_counter.system().label("increment"))
+    .add_system(print_every_other_time.system().after("increment"))
+```
+
+There _are_ some cases where ambiguities are _not_ a bug, such as operations on unordered collection like `Assets`. This is why we don't enable the detector by default. You are free to just ignore these ambiguities, but if you want to suppress the messages in the detector (without defining a dependency), you can add your systems to an "ambiguity set":
+
+```rust
+app
+    .add_system(a.system().in_ambiguity_set("foo"))
+    .add_system(b.system().in_ambiguity_set("foo"))
+```
+
+I want to stress that this is totally optional. Bevy code should be ergonomic and "fun" to write. If sprinkling ambiguity sets everywhere isn't your cup of tea, just don't worry about it!
+
+We are also actively seeking feedback on the new executor. We believe that the new implementation is easier to understand and encourages self-documenting code. The improved parallelism is also nice! But we want to hear from users (both new users starting fresh and old users porting their codebases to the new executor). This space is all about design tradeoffs and feedback will help us ensure we made the right calls.
+
+## Reliable change detection
+
+<div class="release-feature-authors">authors: @Davier, @bjorn3, @alice-i-cecile, @cart</div>
+
+Global change detection, the ability to run queries on the Changed/Added status of any ECS component or resource, just got a major usability boost: changes are now detected across frames/updates:
+
+```rust
+// This is still the same change detection api we all know and love,
+// the only difference is that it "just works" in every situation.
+fn system(query: Query<Entity, Changed<A>>) {
+    // iterates all entities whose A component has changed since
+    // the last run of this system 
+    for e in query.iter() {
+    }
+}
+```
+
+Global change detection was already a feature that set Bevy apart from other ECS frameworks, but now it is completely "fool proof". It works as expected regardless of system ordering, stage membership, or system run criteria.
+
+The old behavior was "systems detect changes that ocurred in systems that ran before them this frame". This was because we used a `bool` to track when each component/resource is added/modified. This flag was cleared for each component at the end of the frame. As a result, users had to be very careful about order of operations, and using features like "system run criteria" could result in dropped changes if systems didn't run on a given update.
+
+We now use a clever "world tick" design that allows systems to detect changes that happened at _any_ point in time since their last run.
```

Comment:
Might be fun to mention *just* how long we can wait between runs to impress the magnitude of their reliability.

Reply:
Hmm i think the implied "it just works" is easier to convey and more approachable than something like "It is reliable up to a tick difference between last change and the last system run of 3 * u32::MAX/4 ticks, where each system run (and each frame) bumps the world tick by 1". While "true" and impressive, I think it muddles the message rather than driving it home.